application/models/services/SurveyUpdater/GeneralSettings.php

```diff
@@ -90,6 +94,19 @@ public function update($surveyId, $input)
             throw new NotFoundException();
         }
 
+        // Before setting the owner, check if the user exists and can be seen
+        // by the current user (in case the request was forged)
+        // NOTE: Internally, the getUserList function will use objects (like the Yii App and Permission model) that
+        //       currently may differ from the ones injected in this service.
+        if (!empty($input['owner_id'])) {
+            $owner = $this->modelUser->findByPk($input['owner_id']);
+            if (!isset($owner) || !in_array($input['owner_id'], getUserList('onlyuidarray'))) {
```

Comment:
On Service ? Not in controller or model ?

Reply:
Is a new pattern being implemented for placing all LS logic outside of framework, on own files.
This is the new "controller"-service on master. That's why we are adding it here.

Reply:
OK :) 

Some other old function need to be move in Service then (DEV issue not needed)

Reply:
I guess that refactoring will slowly be happening with time. :)